specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/stable/2020-03-01/outputs.json

```diff
@@ -0,0 +1,1066 @@
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/EventHubOutputDataSourceProperties",
+          "description": "The properties that are associated with an Event Hub output. Required on PUT (CreateOrReplace) requests.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "EventHubOutputDataSourceProperties": {
+      "description": "The properties that are associated with an Event Hub output.",
+      "allOf": [
+        {
+          "$ref": "./inputs.json#/definitions/EventHubDataSourceProperties"
+        }
+      ],
+      "properties": {
+        "partitionKey": {
+          "type": "string",
+          "description": "The key/column that is used to determine to which partition to send event data."
+        },
+        "propertyColumns": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "AzureSqlDatabaseOutputDataSource": {
+      "description": "Describes an Azure SQL database output data source.",
+      "x-ms-discriminator-value": "Microsoft.Sql/Server/Database",
+      "allOf": [
+        {
+          "$ref": "#/definitions/OutputDataSource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/AzureSqlDatabaseOutputDataSourceProperties",
+          "description": "The properties that are associated with an Azure SQL database output. Required on PUT (CreateOrReplace) requests.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "AzureSqlDatabaseOutputDataSourceProperties": {
+      "description": "The properties that are associated with an Azure SQL database output.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AzureSqlDatabaseDataSourceProperties"
+        }
+      ],
+      "properties": {}
+    },
+    "AzureSynapseOutputDataSource": {
+      "description": "Describes an Azure Synapse output data source.",
+      "x-ms-discriminator-value": "Microsoft.Sql/Server/DataWarehouse",
+      "allOf": [
+        {
+          "$ref": "#/definitions/OutputDataSource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/AzureSynapseOutputDataSourceProperties",
+          "description": "The properties that are associated with an Azure Synapse output. Required on PUT (CreateOrReplace) requests.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "AzureSynapseOutputDataSourceProperties": {
+      "description": "The properties that are associated with an Azure Synapse output.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AzureSynapseDataSourceProperties"
+        }
+      ],
+      "properties": {}
+    },
+    "AzureSynapseDataSourceProperties": {
+      "description": "The properties that are associated with an Azure SQL database data source.",
+      "properties": {
+        "server": {
```

Comment:
same comment as the SQLReference data source.  Why not a resource ID and how do you know how to connect purely based on a server name and database?